### PR TITLE
[Merged by Bors] - chore(*.ModEq): reduce dependencies 

### PR DESCRIPTION
**Pull request by mattrobball**
Both `Nat.ModEq` and `Int.ModEq` have some spurious dependencies which can be removed by reasonable swaps of generic tactics/declarations for `Nat` and `Int` native versions (e.g. use `omega` more). This PR does that and makes a slouching attempt to hold back the import graph downstream of these.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
Looks great. One less item on my todo list.

Note that we could push everything back even more if we reverted to a notation typeclass `Abs` and proved properties of `abs` for `Int` specifically.

File: Mathlib/Data/Nat/ModEq.lean [hunk 1]
```diff
@@ -221,15 +222,22 @@ theorem gcd_eq (h : a ≡ b [MOD m]) : gcd a m = gcd b m := by
     dvd_antisymm (dvd_gcd ((h.dvd_iff h1).mp (gcd_dvd_left a m)) h1)
       (dvd_gcd ((h.dvd_iff h2).mpr (gcd_dvd_left b m)) h2)
 
+lemma _root_.Int.eq_zero_of_abs_lt_dvd {m x : ℤ} (h1 : m ∣ x) (h2 : |x| < m) : x = 0 := by
+  by_contra h
+  have := Int.natAbs_le_of_dvd_ne_zero h1 h
+  rw [Int.abs_eq_natAbs] at h2
+  omega
```
[the remark this answers is not in the document]
Comment (mattrobball) [reply Mathlib/Data/Nat/ModEq.lean]: Yes, these both have a better home there. Thanks.

**Review by YaelDillies — APPROVED** [review]
All looking good now!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**